### PR TITLE
fixed* memory leak

### PR DESCRIPTION
**Pull request by Gestalt-R**
Edit: Read to the very end, it gets better!
#369 
Arghh.. I'm tired and I let a leak slip through. While I do free the tables, it doesn't get rid of the rest of the command. As it is, it leaks a lot. This pull should fix it. The memory leak is exactly the size of all of the user input combined up to that point right now. 

This PR makes the only stored info the stuff we need for the new entries in the hashtable.

That minimizes leaking, but it's still there... technically. The memory is actively useful until the context struct is gotten rid of, which is good enough for the presentation. 

Really, this should be a backlog issue. In order to properly dispose of memory with the name command, I should have either rewritten the init_table or whatever it's called to use malloc, or add a linked list of pointers to be freed to the context struct. 

The best way to get rid of the leak entirely, if 0 prejudice is wanted, is this PR, but then deleting the entire name_operation, and any function call that references it. (Not too much to do.)

### CODE REVIEW

**Comment by Gestalt-R** [comment]
I thoroughly apologize for rushing the pull request #369. You can also revert it if you don't want to bother with this. My guilty conscience wouldn't let me leave it alone after I realized Mark was right. I was too tired to realize I hadn't created the free function for the hashtable with everything in mind.

**Comment by Gestalt-R** [comment]
Most likely, nobody will review this before the presentation. I hope nobody gets screwed over by my bad decision. @borjasotomayor @mpcsh @lydiafilipe

**Comment by Gestalt-R** [comment]
Ignore what I said earlier. This resolves all the leaks.

**Comment by Gestalt-R** [comment]
I said that assuming this commit here was pushed. Sorry.

**Comment by Gestalt-R** [comment]
Like seriously, there are not even "necessary" leaks. They are all gone.

File: src/cli/src/cmd.c [hunk 1]
```diff
@@ -14,7 +14,9 @@
 void add_entry(char *command_name, operation *associated_operation, action_type_t *action, lookup_t **table)
 {
     lookup_t *t = malloc(sizeof(lookup_t));
-    t->name = command_name;
+    char * newname = malloc(sizeof(char) * strlen(command_name));
```
Comment (mpcsh) [inline src/cli/src/cmd.c]: Should be `char *newname`

File: src/ui/src/print_functions.c [hunk 2]
```diff
@@ -126,6 +126,11 @@ void print_cli(chiventure_ctx_t *ctx, window_t *win)
         do_cmd(c, &quit, ctx);
     }
 
+
```
Comment (mpcsh) [inline src/ui/src/print_functions.c]: This extra newline should be deleted

Comment (Gestalt-R) [reply src/ui/src/print_functions.c]: Did so.

File: src/cli/src/cmd.c (hunk 1 above)
Comment (Gestalt-R) [reply src/cli/src/cmd.c]: Yup.

**Comment by Gestalt-R** [comment]
This should not be put into master until after the presentation. I consider it very useful for anyone continuing to work on this project.

**Pull request closed without merging**